Updates docs for CLI, FAQ, and QuickStart (#1755)
redis/redis-doc: docs/getting-started/faq.md

## I like Redis's high level operations and features, but I don't like that it keeps everything in memory and I can't have a dataset larger than memory. Are there any plans to change this?

In the past the Redis developers experimented with Virtual Memory and other systems in order to allow larger than RAM datasets, but after all we are very happy if we can do one thing well: data served from memory, disk used for storage. So for now there are no plans to create an on disk backend for Redis. Most of what
Redis is, after all, a direct result of its current design.

If your real problem is not the total RAM needed, but the fact that you need
to split your data set into multiple Redis instances, please read the
[Partitioning page](/topics/partitioning) in this documentation for more info.

> Recently Redis LLC, the company sponsoring Redis developments, developed a

Reviewer: ```suggestion
Redis Ltd., the company sponsoring Redis development, has developed a
```

> "Redis on Flash" solution that is able to use a mixed RAM/flash approach for

Reviewer: ```suggestion
"Redis on Flash" solution that uses a mixed RAM/flash approach for
```

And now the long one:

> The Redis background saving schema relies on the copy-on-write semantic of fork in

Reviewer: ```suggestion
The Redis background saving schema relies on the copy-on-write semantic of the `fork` system call in
```

You can find more information about using multiple Redis instances in the [Partitioning page](/topics/partitioning).

> As of version 4.0 Redis has started implementing threaded actions. For now this is limited to deleting objects in the background and blocking commands implemented via Redis modules. For subsequent releases, the plan is to make Redis more and more threaded.

Reviewer: ```suggestion
As of version 4.0, Redis has started implementing threaded actions. For now this is limited to deleting objects in the background and blocking commands implemented via Redis modules. For subsequent releases, the plan is to make Redis more and more threaded.
```

> ## What is the maximum number of keys a single Redis instance can hold? What is the max number of elements in a Hash, List, Set, Sorted Set?

Reviewer: ```suggestion
## What is the maximum number of keys a single Redis instance can hold? What is the maximum number of elements in a Hash, List, Set, and Sorted Set?
```

If you use keys with limited time to live (Redis expires) this is normal behavior. This is what happens:

* The primary generates an RDB file on the first synchronization with the replica.
* The RDB file will not include keys already expired in the primary but are still in memory.

> * These keys are still in the memory of the Redis primary, even if logically expired. They'll not be considered as existing  but the memory will be reclaimed later - both incrementally and explicitly on access. While these keys are not logically part of the dataset, they are advertised in `INFO` output and by the `DBSIZE` command.

Reviewer: ```suggestion
* These keys are still in the memory of the Redis primary, even if logically expired. They'll be considered non-existent, and their memory will be reclaimed later, either incrementally or explicitly on access. While these keys are not logically part of the dataset, they are accounted for in the `INFO` output and in the `DBSIZE` command.
```

* When the replica reads the RDB file generated by the primary, this set of keys will not be loaded.

> As a result of this, it is common for users with many keys with an expire set to see less keys in the replicas. Because of this artifact, there is no actual logical difference in the instances content.

Reviewer: ```suggestion
As a result of this, it is common for users with many keys with an expire set to see fewer keys in the replicas. Because of this artifact, there is no actual logical difference in the instance's content.
```
